src/Transport/Receiving/MessagePump.cs

```diff
@@ -322,33 +327,54 @@ await processor.CloseAsync(cancellationToken)
                     {
                         await processMessageEventArgs.SafeAbandonMessageAsync(message,
                                 transportSettings.TransportTransactionMode,
-                                cancellationToken: messageProcessingCancellationToken)
+                                cancellationToken: processingTokenSource.Token)
                             .ConfigureAwait(false);
                     }
                 }
-                catch (ServiceBusException onErrorEx) when (onErrorEx.IsTransient || onErrorEx.Reason is ServiceBusFailureReason.MessageLockLost)
+                catch (ServiceBusException onErrorEx) when (onErrorEx.IsTransient ||
+                                                            onErrorEx.Reason is ServiceBusFailureReason.MessageLockLost)
                 {
                     Logger.Debug("Failed to execute recoverability.", onErrorEx);
 
                     await processMessageEventArgs.SafeAbandonMessageAsync(message,
                             transportSettings.TransportTransactionMode,
-                            cancellationToken: messageProcessingCancellationToken)
+                            cancellationToken: processingTokenSource.Token)
                         .ConfigureAwait(false);
                 }
-                catch (Exception onErrorEx) when (onErrorEx.IsCausedBy(messageProcessingCancellationToken))
+                catch (Exception onErrorEx) when (onErrorEx.IsCausedBy(processingTokenSource.Token))
                 {
                     throw;
                 }
                 catch (Exception onErrorEx)
                 {
-                    criticalErrorAction($"Failed to execute recoverability policy for message with native ID: `{message.MessageId}`", onErrorEx, messageProcessingCancellationToken);
+                    criticalErrorAction(
+                        $"Failed to execute recoverability policy for message with native ID: `{message.MessageId}`",
+                        onErrorEx, processingTokenSource.Token);
```

Comment:
This is the place where I wasn't sure what token to add. You could argue that here it might make sense to add the `messageProcessingCancellationToken` so that the critical error action is only cancelled when the endpoint shuts down.